### PR TITLE
refactor : ddd cqrs

### PR DESCRIPTION
**Pull request by wzh425**
1、refactor: ChannelType use Enumeration,and provide different implementations ，kill the switch(src/Services/Masa.Mc.Service/Domain/Channels/Aggregates/ChannelType.cs)
2、refactor: 发送前可以编辑的收件人数据冗余，解析后的收件人数据不冗余，只保存发消息需要的渠道用户数据
3、refactor: api调用发送消息分开成外部用户和内部用户
4、refactor:cqrs，建立查询模型和上下文，查询事件使用查询上下文
4、refactor: ddd  聚合根中抽出一些值对象，应用层业务迁移到领域服务、聚合根、值对象、基础设施内（25%，做了业务代码迁移聚合根内部还未好好提炼）
5、refactor: 任务收件人数据解析成消息发送收件人数据 实现迁移到基础设施层，提供程序为Auth，需要调用外部的事件处理上层到应用层

### CODE REVIEW

File: src/Infrastructure/Masa.Mc.Infrastructure.Common/Utils/Check.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+public static class Check
```
Comment (doddgu) [inline src/Infrastructure/Masa.Mc.Infrastructure.Common/Utils/Check.cs:1]: MASA Framework has this class. You can ask zhenlei.

File: src/Infrastructure/Masa.Mc.Infrastructure.EntityFrameworkCore/Extensions/QueryableExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
+﻿namespace Masa.Mc.Infrastructure.EntityFrameworkCore.EntityFrameworkCore.Extensions;
+
+public static class QueryableExtensions
```
Comment (doddgu) [inline src/Infrastructure/Masa.Mc.Infrastructure.EntityFrameworkCore/Extensions/QueryableExtensions.cs:3]: I think it need to move to MASA.Framework.

File: src/Services/Masa.Mc.Service/Domain/WebsiteMessages/EventHandler/WebsiteMessageCreatedEventHandler.cs [hunk 3]
```diff
@@ -41,22 +41,14 @@ public async Task HandleEvent(AddWebsiteMessageDomainEvent @event)
...
-            {
-                UserId = currentUser.Id,
-                DisplayName = currentUser.DisplayName ?? string.Empty,
-                Account = currentUser.Account,
-                Email = currentUser.Email ?? string.Empty,
-                PhoneNumber = currentUser.PhoneNumber ?? string.Empty
-            };
 
-            var messageRecord = new MessageRecord(receiverUser.UserId, taskHistory.MessageTask.ChannelId.Value, taskHistory.MessageTaskId, taskHistory.Id, taskHistory.MessageTask.Variables, messageData.GetDataValue<string>(nameof(MessageTemplate.Title)), taskHistory.SendTime);
+            //var receiver = new Receiver(currentUser.Id, currentUser.DisplayName, currentUser.Avatar, currentUser.PhoneNumber, currentUser.Email);
```
Comment (doddgu) [inline src/Services/Masa.Mc.Service/Domain/WebsiteMessages/EventHandler/WebsiteMessageCreatedEventHandler.cs:45]: ?

File: src/Services/Masa.Mc.Service/Infrastructure/EntityFrameworkCore/McQueryContext.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            b.ToView(MCConsts.DbTablePrefix + "MessageReceiverUsers", MCConsts.DbSchema);
+            b.Property(x => x.Variables).HasConversion(new ExtraPropertiesValueConverter()).Metadata.SetValueComparer(new ExtraPropertyDictionaryValueComparer());
+        });
+
+        builder.Entity<WebsiteMessageQueryModel>(b =>
+        {
+            b.ToView(MCConsts.DbTablePrefix + "WebsiteMessages", MCConsts.DbSchema);
+        });
+
+        //builder.Entity<WebsiteMessageCursor>(b =>
```
Comment (doddgu) [inline src/Services/Masa.Mc.Service/Infrastructure/EntityFrameworkCore/McQueryContext.cs:94]: ?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Services/Masa.Mc.Service/Domain/WebsiteMessages/EventHandler/WebsiteMessageCreatedEventHandler.cs (hunk 3 above)
Comment (wzh425) [reply src/Services/Masa.Mc.Service/Domain/WebsiteMessages/EventHandler/WebsiteMessageCreatedEventHandler.cs:45]: This is not the latest code. Message records do not require redundant additional user information, but only userid and channel user ID

File: src/Services/Masa.Mc.Service/Infrastructure/EntityFrameworkCore/McQueryContext.cs (hunk 4 above)
Comment (wzh425) [reply src/Services/Masa.Mc.Service/Infrastructure/EntityFrameworkCore/McQueryContext.cs:94]: 这是为查询上下文的ef配置，用于查询事件，如果读写不分离  建立领域模型时还要兼顾查询场景下 会让建立聚合根时非常难受

Comment (wzh425) [reply src/Services/Masa.Mc.Service/Infrastructure/EntityFrameworkCore/McQueryContext.cs:94]: 查询事件里 没有这个站内行游标  他在命令事件下才需要查询

Comment (wzh425) [reply src/Services/Masa.Mc.Service/Infrastructure/EntityFrameworkCore/McQueryContext.cs]: Most of the time, two models should be in one aggregation root, but for the convenience of query, they are separated into two aggregation roots, which makes me very uncomfortable

File: src/Infrastructure/Masa.Mc.Infrastructure.Common/Utils/Check.cs (hunk 1 above)
Comment (wzh425) [reply src/Infrastructure/Masa.Mc.Infrastructure.Common/Utils/Check.cs:1]: MasaArgumentException. ThrowIfNull is not marked with [NotNull], which cannot eliminate the nullable prompt

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**